core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilterWithJwt.java

```diff
@@ -77,4 +89,44 @@ protected void validateAccessToken(String cookieValue)
         assertThat(claims.get("client_id")).isEqualTo(TRINO_CLIENT_ID);
         assertThat(claims.getIssuer()).isEqualTo("https://localhost:4444/");
     }
+
+    @Test
+    public void testBearerAuthenticatorUserMappingApplied()
```

Comment:
Is this a valid use-case?
What's wrong with using `X-Trino-User=TRINO_CLIENT_USER` in the first place? Or even better with using a principal claim containing `TRINO_CLIENT_USER`?
IIRC the motivation to add this header rewrite function was to enable user mapping for clients not supporting `--session-user` configuration property. For them password authentication with user mapping just didn't work at all, hence this hack. What's different for OAuth2 is that users can still set `--user` as it'll work just fine. Might not be the best user experience (arguably) but it's not a blocker.
Additionally, what about other authenticators? If this change makes sense for OAuth2 does it also make sense for Kerberos etc.?

Reply:
Purpose of this PR was to align behaviour from PasswordAuthenticator to OAuth2 https://github.com/trinodb/trino/commit/bcf6c581dea7c9b1b9c678e6f867510fa51ff63d
And as i understand it is a kind of hack :) but I can't tell for sure.
If i understand well this might be applied for any kind of authentication where credentials (username mostly) is being mapped

Reply:
After offline discussion with @lukasz-walkiewicz we agreed that this is ok to add this behaviour for use cases in which IdP returns Subject in token claims in other case that we internally use to verify access controls.

I've also added 2 additional test cases suggested by Lukasz.